client/src/pages/dashboard/views/Invites.svelte

```diff
@@ -1,12 +1,77 @@
+    import InviteRow from '../../../components/ui/itemrow/InviteRow.svelte';
+    import Modal from '../../../components/ui/Modal.svelte';
+    import PersonAdd from '../../../components/icons/PersonAdd.svelte';
+    import { Invite } from '../../../api/invite.ts';
+    import { RowType, InvitePayload, Events, IconColor, IconSize } from '../../../components/types.ts';
+    import { assert } from '../../../assert.ts';
     import { dashboardState } from '../stores/DashboardState';
+    import { inviteList } from '../../../pages/dashboard/stores/InviteStore.ts';
 
+    let showInviteForm = false;
+    let showRevokeInviteContextMenu = false;
+    let currentContext = null as InvitePayload | null;
     $: ({ currentOffice } = $dashboardState);
+
+    function overflowClickHandler(e: CustomEvent<InvitePayload>) {
+        currentContext = e.detail;
+        showRevokeInviteContextMenu = currentContext.ty === RowType.Invite;
+    }
+
+    async function removeInviteHandler(e: CustomEvent<InvitePayload>) {
+        switch (e.type) {
+            case Events.RemoveInvitation:
+                assert(currentContext !== null);
+                assert(e.detail.email !== 'undefined');
+                try {
+                    await Invite.revoke({
+                        office: currentContext.office,
+                        email: e.detail.email,
+                    });
+                    await inviteList.reload?.();
+                } catch (err) {
+                    alert(err);
+                }
+                break;
+            default: break;
+        }
+    }
 </script>
 
 {#if currentOffice === null}
     You must select an office before accessing the Invites page.
 {:else}
-    Invites page of Office ID {currentOffice}.
-    <h1>Invites</h1>
+    <Button on:click={() => (showInviteForm = true)}>
+        <PersonAdd color={IconColor.Default} size={IconSize.Normal} alt="Invite person" />Invite User
+    </Button>
+    <Modal title="Invite User" bind:showModal={showInviteForm}>
+        {#if $dashboardState.currentOffice === null}
+            <span>No selected office.</span>
+        {:else}
+            <InviteForm />
+        {/if}
+    </Modal>
+    {#if typeof $inviteList === 'undefined' || $inviteList.length === 0}
+            <h3>No invite backlogs, yay!</h3>
+    {:else}
+        {#each $inviteList as { email, permission, creation } (email)}
+            <InviteRow
+                email={email}
+                office={$dashboardState.currentOffice ?? 0}
+                permission={permission}
+                iconSize={IconSize.Large}
+                creation={creation}
+                on:overflowClick={overflowClickHandler}
+            />
+        {/each}
+        {#if currentContext?.ty === RowType.Invite}
+            <InviteContext
+                bind:show={showRevokeInviteContextMenu}
+                payload={currentContext}
+                on:removeInvitation={removeInviteHandler} />
+        {/if}

```

Comment:
```suggestion
<Modal title="Revoke Invite" bind:showModal={showRevokeModal}}>
    <RevokeInvite />
</Modal>
```
We don't move the entire form to the page. We import it and wrap it inside a Modal.
You can see examples in `Inbox.svelte` and other pages.

Reply:
My intent was using the modal (it was more familiar), actually. But I think I just followed the row template that you guys did for inbox.

Reply:
![image](https://user-images.githubusercontent.com/68340609/235835201-45f26d37-8b34-4669-a3c1-385347737397.png)

Reply:
Okay, I see what happened. Thank you!

Reply:
Okay! I realize what the intention was now. I recommend removing the "You are removing the invitation" message and adding it in a separate modal to keep the consistency of the overflow menu.